### PR TITLE
Add Update Progress Indicator

### PR DESCRIPTION
**Pull request by alainm23**
Fixed https://github.com/elementary/switchboard-plug-about/issues/310

### CODE REVIEW

**Review by stsdc — APPROVED** [review]

File: src/Backends/SystemUpdate.vala [hunk 1]
```diff
@@ -189,7 +191,8 @@ public class SettingsDaemon.Backends.SystemUpdate : Object {
     }
 
     private void progress_callback (Pk.Progress progress, Pk.ProgressType progress_type) {
-        update_state (current_state.state, PkUtils.status_to_title (progress.status));
+        update_state (current_state.state, PkUtils.status_to_title (progress.status),
```
Comment (leolost2605) [inline src/Backends/SystemUpdate.vala:194]: If you have to split arguments into lines, a new line for every argument is preferred, see https://docs.elementary.io/develop/writing-apps/code-style#splitting-arguments-into-lines

File: src/Backends/SystemUpdate.vala [hunk 2]
```diff
@@ -203,10 +206,13 @@ public class SettingsDaemon.Backends.SystemUpdate : Object {
         update_state (ERROR, message);
     }
 
-    private void update_state (PkUtils.State state, string message = "") {
+    private void update_state (PkUtils.State state, string message = "",
```
Comment (leolost2605) [inline src/Backends/SystemUpdate.vala:209]: Same here

File: src/Backends/UbuntuDrivers.vala [hunk 3]
```diff
@@ -260,7 +262,8 @@ public class SettingsDaemon.Backends.UbuntuDrivers : Object {
     }
 
     private void progress_callback (Pk.Progress progress, Pk.ProgressType progress_type) {
-        update_state (current_state.state, PkUtils.status_to_title (progress.status));
+        update_state (current_state.state, PkUtils.status_to_title (progress.status),
```
Comment (leolost2605) [inline src/Backends/UbuntuDrivers.vala:265]: And here

File: src/Backends/UbuntuDrivers.vala [hunk 4]
```diff
@@ -274,10 +277,13 @@ public class SettingsDaemon.Backends.UbuntuDrivers : Object {
         update_state (ERROR, message);
     }
 
-    private void update_state (PkUtils.State state, string message = "") {
+    private void update_state (PkUtils.State state, string message = "",
+        uint percentage = 0, uint64 download_size_remaining = 0) {
```
Comment (leolost2605) [inline src/Backends/UbuntuDrivers.vala:281]: And here

**Review by leolost2605 — CHANGES_REQUESTED** [review]

**Review by leolost2605 — APPROVED** [review]
LGTM

**Pull request merged**